### PR TITLE
build: Update the release machinery

### PR DESCRIPTION
**Pull request by flub**
This should be able to use modern release automation now.

### CODE REVIEW

File: .craft.yml [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
 changelogPolicy: simple
+preReleaseCommand: bash scripts/bump-version
 statusProvider:
   name: github
 artifactProvider:
-  name: none
+  name: github
+requireNames:
+  - /^.*.zip$/
+  - /^.*.tar.gz$/
```
Comment (jan-auer) [inline .craft.yml:18]:
```suggestion
  - /^.*\.zip$/
  - /^.*\.tar\.gz$/
```

File: .craft.yml [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
+minVersion: "0.23.1"
+changelogPolicy: auto
 github:
   owner: getsentry
   repo: rust-debugid
 targets:
   - name: crates
   - name: github
 changelogPolicy: simple
+preReleaseCommand: bash scripts/bump-version
```
Comment (jan-auer) [inline .craft.yml:11]:
```suggestion
preReleaseCommand: bash scripts/bump-version.sh
```

**Review by jan-auer — COMMENTED** [review]
While we're on it, we should run CI (tests and linters) in GitHub rather than Travis. The GitHub status provider should report this in the meanwhile, can you confirm?

**Comment by flub** [comment]
> While we're on it, we should run CI (tests and linters) in GitHub rather than Travis.

I thought about this but it works so left it alone (never break working systems at all).  Is a policy or so that we should move things?

> The GitHub status provider should report this in the meanwhile, can you confirm?

It works on PRs so I assume that's what you mean?  Otherwise I have no idea.

**Review by jan-auer — APPROVED** [review]

**Pull request merged**